kimchi/src/circuits/polynomials/foreign_field_add/circuitgates.rs

```diff
@@ -153,31 +157,30 @@ where
             (sign.clone() + T::one()) * (sign.clone() - T::one()),
         ];
 
-        // Constraints to check the carry flags are -1, 0, or 1.
-        checks.push(carry(&carry_lo));
-        checks.push(carry(&carry_mi));
+        // Constraints to check the carry flag is -1, 0, or 1.
+        checks.push(is_carry(&carry));
+        //checks.push(carry(&carry_mi));
 
         // Auxiliary inline function to obtain the constraints of a foreign field addition result
         let mut result = {
-            // r_0 = a_0 + s * b_0 - q * f_0 - 2^88 * c_0
-            let result_computed_lo = left_input_lo + sign.clone() * right_input_lo
-                - field_overflow.clone() * foreign_modulus[0].clone()
-                - carry_lo.clone() * two_to_limb.clone();
-            // r_1 = a_1 + s * b_1 - q * f_1 - 2^88 * c_1 + c_0
-            let result_computed_mi = left_input_mi + sign.clone() * right_input_mi
-                - field_overflow.clone() * foreign_modulus[1].clone()
-                - carry_mi.clone() * two_to_limb
-                + carry_lo;
-            // r_2 = a_2 + s * b_2 - q * f_2 + c_1
-            let result_computed_hi = left_input_hi + sign * right_input_hi
+            // a_bot = a_0 + a_1 * 2^88
+            // b_bot = b_0 + b_0 * 2^88
+            // f_bot = f_0 + f_1 * 2^88
+            // r_bot = a_bot + s * b_bot - q * f_bot - c * 2^176
+            let result_bot = two_limb(&left_input_lo, &left_input_mi)
+                + sign.clone() * two_limb(&right_input_lo, &right_input_mi)
+                - field_overflow.clone() * two_limb(&foreign_modulus[0], &foreign_modulus[1])
+                - carry.clone() * two_to_limb.clone() * two_to_limb;
```

Comment:
Instead of `two_to_limb.clone() * two_to_limb` you can now just do

`T::two_to_2limb()`